examples/golang/simpleinterconnectprovider/handler/handler.go

```diff
@@ -99,39 +168,227 @@ func (p *PrototypeHandler) DeleteTransceiver(_ context.Context, trans *pb.Delete
+	}
+
+	return p.bearers[bearer.Name], nil
 }
-func (p *PrototypeHandler) DeleteBearer(context.Context, *pb.DeleteBearerRequest) (*emptypb.Empty, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method DeleteBearer not implemented")
+
+func (p *PrototypeHandler) CreateBearer(_ context.Context, bearer *pb.CreateBearerRequest) (*pb.Bearer, error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	bearerName := fmt.Sprintf("bearers/%s", bearer.BearerId)
+	if p.bearers[bearerName] != nil {
+		return nil, status.Errorf(codes.AlreadyExists, "bearer with requested ID was already created")
+	}
+	if bearer.Bearer.Interval.StartTime.AsTime().After(bearer.Bearer.Interval.EndTime.AsTime()) {
+		return nil, status.Errorf(codes.InvalidArgument, "bearer has negative time interval argument")
+	}
+	if !p.checkForSufficientContactWindow(bearer.Bearer) {
+		return nil, status.Errorf(codes.FailedPrecondition, "bearer has no sufficient contact window")
+	}
+	// Override the name of the attachment circuit to ensure that it has the correct resource name.
+	// It is up to the API to either validate the correctness of the name or just override it on creation.
+	bearer.Bearer.Name = bearerName
+
+	p.bearers[bearerName] = bearer.Bearer
+
+	return bearer.Bearer, nil
 }
+
+func (p *PrototypeHandler) checkForSufficientContactWindow(bearer *pb.Bearer) bool {
+	for _, contactWindow := range p.contactWindows {
+		if contactWindow.Target != bearer.Target || contactWindow.Transceiver != bearer.Transceiver {
+			continue
+		}
+
+		if contactWindow.Interval.StartTime.AsTime().After(bearer.Interval.StartTime.AsTime()) ||
+			contactWindow.Interval.EndTime.AsTime().Before(bearer.Interval.EndTime.AsTime()) {
+			continue
+		}
+
+		if contactWindow.MinRxCenterFrequencyHz >= bearer.RxCenterFrequencyHz ||
+			contactWindow.MaxRxCenterFrequencyHz <= bearer.RxCenterFrequencyHz ||
+			contactWindow.MinRxBandwidthHz >= bearer.RxBandwidthHz ||
+			contactWindow.MaxRxBandwidthHz <= bearer.RxBandwidthHz ||
+			contactWindow.MinTxCenterFrequencyHz >= bearer.TxCenterFrequencyHz ||
+			contactWindow.MaxTxCenterFrequencyHz <= bearer.TxCenterFrequencyHz ||
+			contactWindow.MinTxBandwidthHz >= bearer.TxBandwidthHz ||
+			contactWindow.MaxTxBandwidthHz <= bearer.TxBandwidthHz {
+			continue
+		}
+
+		for _, knownBearer := range p.bearers {
+			if bearer.Target != knownBearer.Target || bearer.Transceiver != knownBearer.Transceiver {
+				continue
+			}
+
+			if math.Abs(float64(knownBearer.RxCenterFrequencyHz-bearer.RxCenterFrequencyHz))-(float64(knownBearer.RxBandwidthHz)+float64(bearer.RxBandwidthHz))/2 > 0 &&
+				math.Abs(float64(knownBearer.TxCenterFrequencyHz-bearer.TxCenterFrequencyHz))-(float64(knownBearer.TxBandwidthHz)+float64(bearer.TxBandwidthHz))/2 > 0 {
+				return true
+			}
+
+			if knownBearer.Interval.EndTime.AsTime().After(bearer.Interval.StartTime.AsTime()) && bearer.Interval.StartTime.AsTime().After(knownBearer.Interval.StartTime.AsTime()) ||
```

Comment:
I believe there may be a clarity issue on Interval use for bearer's and ContactWindows - is it ever documented whether they are closed intervals [start, stop] or half-open [start, stop) etc?  As it impacts this logic and generally I think we probably want half-open so that you can chain them together and they don't overlap?  Maybe an issue for the Proto comments (unless Intervals have already defined this as Google defines them? Maybe they have).

Reply:
Correct. It is half open [start,stop) so I should be more careful here. I'll fix that and I'll add a comment somewhere (everywhere?) in the API so that people don't have to look it up.

Reply:
Although, I THINK my logic is fine. I added a test for time adjacent bearers and that works correctly, so aside from comments we should hopefully be fine. I can't think of a different edge case right now.